core/cache/cache.go

```diff
@@ -28,6 +28,8 @@ type (
 		Purge(key string) error
 		PurgeTags(tags []string) error
 		Flush() error
+		TagSupport() bool
+		FlushSupport() bool
```

Comment:
Same as above. Also I would not add this at all. Just call `Flush`. If the backend doesn't support it, the backend can ignore it. The interface wants a `Flush` method which makes sense for caches, so we should leave this to the interface and not the users of the interface to decide what happens.

```
func (*B) FlushSupport() bool { return false }
func (*B) Flush() { }
// ...
if backend.FlushSupport() {
    backend.Flush()
}
```
is pretty much the same as
```
func (*B) Flush() { }
// ...
backend.Flush()
```
however in the later case the compiler can optimize and we omit one function call and branch.

Reply:
ok. i change the interface Backend to it's original version and a added a new interface TagSupportingBackend.

But this has currently one caveat: The multilevel backend is not able to support tags. for the moment this is ok. we could allow to add TagSupportingBackends into MultiLevelBackendOptions to solve this. But i don't think this would be a future proof-solution, because we don't know if we would have another type of backend-features.
And we could allow to add an interface{} to MultiLevelBackendOptions, but this would break the type-safety.

please let me now, if you have an prefered option here.